File: weather-app/app/src/main/java/com/rperazzo/weatherapp/view/adapters/FindItemAdapter.java
package com.rperazzo.weatherapp.view.adapters;

import android.content.Context;
import android.content.SharedPreferences;
import android.view.LayoutInflater;
import android.view.View;
import android.view.ViewGroup;
import android.widget.ArrayAdapter;
import android.widget.ImageView;
import android.widget.TextView;

import com.rperazzo.weatherapp.R;
import com.rperazzo.weatherapp.model.City;
import com.rperazzo.weatherapp.viewmodel.WeatherViewModel;

import java.util.ArrayList;

public class FindItemAdapter extends ArrayAdapter<City> {

    WeatherViewModel viewModel;

    public FindItemAdapter(Context context, ArrayList<City> cities, WeatherViewModel viewModel) {
        super(context, 0, cities);
        this.viewModel = viewModel;
    }

    @Override
    public View getView(int position, View convertView, ViewGroup parent) {

        final City city = getItem(position);

        if (convertView == null) {
            convertView = LayoutInflater
                    .from(getContext())
                    .inflate(R.layout.city_list_item, parent, false);
        }

        TextView cityName = convertView.findViewById(R.id.cityNameTxt);
        TextView description = convertView.findViewById(R.id.descriptionTxt);
        TextView metric = convertView.findViewById(R.id.metricTxt);
        TextView temp = convertView.findViewById(R.id.tempTxt);
        TextView wind = convertView.findViewById(R.id.windTxt);
        TextView clouds = convertView.findViewById(R.id.cloudsTxt);
        TextView pressure = convertView.findViewById(R.id.pressureTxt);
        ImageView icon = convertView.findViewById(R.id.weatherIcon);

        cityName.setText(city.getTitle());
        description.setText(city.getDescription());
        temp.setText(city.getTemperature());
        clouds.setText(city.getClouds());
        pressure.setText(city.getPressure());

        String units = viewModel.getTemperatureUnit();
        if ("metric".equals(units)) {
            metric.setText("ºC");
            wind.setText(city.getWind() + " m/s");
        } else {
            metric.setText("ºF");
            wind.setText(city.getWind() + " m/h");
        }

        int resId = getContext().getResources().getIdentifier(
            "w_"+city.weather.get(0).icon,
            "drawable",
            getContext().getPackageName()
        );

        icon.setImageResource(resId);

        return convertView;
    }

}

File: weather-app/app/src/main/java/com/rperazzo/weatherapp/MVVM/view/adapter/FindItemAdapter.kt
package com.rperazzo.weatherapp.MVVM.view.adapter

import android.content.Context
import android.view.LayoutInflater
import android.view.View
import android.view.ViewGroup
import android.widget.ArrayAdapter
import android.widget.ImageView
import android.widget.TextView
import com.rperazzo.weatherapp.R
import com.rperazzo.weatherapp.MVVM.model.entity.City
import com.rperazzo.weatherapp.MVVM.view.activity.MainActivity

class FindItemAdapter(context: Context, val cities: ArrayList<City>): ArrayAdapter<City>(context, 0, cities) {

    @Override
    override fun getView(position: Int, convertView: View?, parent: ViewGroup): View{
        var viewItem :View;
        val city: City = getItem(position);

        if (convertView == null) {
            viewItem = LayoutInflater.from(getContext())
                    .inflate(R.layout.city_list_item, parent, false);
        }else{
            viewItem = convertView;
        }
        val cityName: TextView = viewItem.findViewById(R.id.cityNameTxt);
        cityName.setText(city.getTitle());

        val description: TextView = viewItem.findViewById(R.id.descriptionTxt);
        description.setText(city.getDescription());

        val metric: TextView = viewItem.findViewById(R.id.metricTxt);
        val units: String = (context as MainActivity).getTemperatureUnit()
        if ("metric".equals(units)) {
            metric.setText("ºC");
        } else {
            metric.setText("ºF");
        }

        val temp: TextView = viewItem.findViewById(R.id.tempTxt);
        temp.setText(city.getTemperature());

        val wind: TextView = viewItem.findViewById(R.id.windTxt);
        if ("metric".equals(units)) {
            wind.setText(city.getWind() + " m/s");
        } else {
            wind.setText(city.getWind() + " m/h");
        }

        val clouds: TextView = viewItem.findViewById(R.id.cloudsTxt);
        clouds.setText(city.getClouds());

        val pressure : TextView = viewItem.findViewById(R.id.pressureTxt);
        pressure.setText(city.getPressure());

        val resId: Int = getContext().getResources().getIdentifier(
                "w_"+city.weather.get(0).icon,
                "drawable",
                getContext().getPackageName());

        val icon: ImageView = viewItem.findViewById(R.id.weatherIcon);
        icon.setImageResource(resId);

        return viewItem;
    }
}